### ISSUE TITLE
next-error conflicts with eldoc in echo area

### ISSUE THREAD

**Issue by ywwry66** [flycheck/flycheck#1822]
This template is for **user support**.  If you are reporting an actual Flycheck bug, please use the relevant issue template instead.

## Checklist
- [x] I have searched existing issues and StackOverflow for solutions.
- [x] I have read the [Flycheck manual][].
- [x] I have read the [Troubleshooting Guide][].
- [x] I have checked my issue has not been reported yet.

## Problem description
I am not sure if this is related to simple.el or flycheck-next-error-function, so I post this as "user support". If I use flycheck-next-error, no matter where the point is, I only get messages from flycheck (no messages from eldoc), which is the expected behavior. However, if I use next-error, I get messages from flycheck for around 0.4 secs then they are overwritten by eldoc message.

## Steps to reproduce
1. Paste (setq foo bar) in the scratch buffer
2. C-a
3. M-x flycheck-next-error / M-x next-error

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior
If I use flycheck-next-error or next-error, I expect to only see the error message from flycheck.

## System configuration
```
Syntax checkers for buffer *scratch* in lisp-interaction-mode:

First checker to run:

  emacs-lisp
    - may enable:    yes
    - may run:       t
    - executable:    Found at d:/Emacs/bin/emacs.exe
    - next checkers: emacs-lisp-checkdoc

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled
checkers.

--------------------

Flycheck version: 32snapshot (package: 20200820.1403)
Emacs version:    27.1
System:           x86_64-w64-mingw32
Window system:    w32
```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes

[Flycheck manual]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by fmdkdd** [comment]
Thanks, I can reproduce the behavior. However, I'm not sure that is a bug.  Looks like `next-error` just moves point to the next error, which is intended.  Notice that if you just move your cursor to `foo` manually, you get the flycheck error message in the area, then eldoc, just as with `next-error`.

I can agree that having the error briefly display is not very useful, but then that's the limit of using the echo area anyway.  I can recommend `flycheck-inline` or other alternatives to `flycheck-display-error-function`.

I'm closing this, but feel free to answer if you feel we should change the default.

**Comment by ywwry66** [comment]
Wasn't aware of the `flycheck-inline` package. It looks great!

I tried moving the cursor manually, but what I got is slightly different. I got the eldoc message first, then the Flycheck error message. 

I posted this issue because I was thinking what default behavior should make more sense, otherwise I would simply bind `M-g n` to `flycheck-next-error` and I would be happy (since `flycheck-next-error` gives me the desired result and `M-g n` is the intuitive key binding). Personally, I think when someone wants to jump to the next error, he also wants to see the error info so that he can fix it. But `next-error` fails to help in this regard. Reading the instructions from [https://www.flycheck.org/en/latest/user/error-interaction.html](url), it seems you recommend `next-error` over `flycheck-next-error`, while I think the latter is way more useful.

**Comment by ywwry66** [comment]
Just tried `flycheck-inline`. It is interesting that `next-error` works better than `flycheck-next-error` in this minor mode. With `next-error`, both Flycheck error message and eldoc message are visible. With `flycheck-next-error`, the eldoc message is suppressed, so that the echo area is not used at all.
